core/shared/src/main/scala/zio/Schedule.scala

```diff
@@ -1126,6 +1127,34 @@ object Schedule {
     Schedule(loop(None, 0L))
   }
 
+  /**
+   * Cron-like schedule that recurs every specified `minute` of each hour.
+   * It triggers at zero second of the minute.
+   * Producing a count of repeats: 0, 1, 2.
+   */
+  def minuteOfHour(minute: Int): UIO[Schedule[Any, Any, Long]] = ZIO.effectTotal {
```

Comment:
This is not acceptable in my view. All the constructors of schedule should return Schedule values. Otherwise composition is too painful. If you really want to check the range, then please make that lazy inside the `loop` call. It will defer the error until the schedule is run, but that is the best we can do if we want to guarantee the errors can be managed from within ZIO.